discovery-client/src/main/java/io/micronaut/discovery/oraclecloud/vault/config/OracleCloudVaultConfigurationClient.java

```diff
@@ -0,0 +1,218 @@
+import io.micronaut.context.env.PropertySource;
+import io.micronaut.discovery.config.ConfigurationClient;
+import io.micronaut.runtime.ApplicationConfiguration;
+import io.micronaut.scheduling.TaskExecutors;
+import io.reactivex.Flowable;
+import io.reactivex.Scheduler;
+import io.reactivex.schedulers.Schedulers;
+import org.apache.commons.codec.binary.Base64;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.ExecutorService;
+
+/**
+ *  A {@link ConfigurationClient} for Oracle Cloud Vault Configuration.
+ *
+ *  @author toddsharp
+ *  @since 2.0.0
+ */
+@Singleton
+@Requires(classes = {
+        SecretsClient.class,
+        VaultsClient.class,
+        InstancePrincipalsAuthenticationDetailsProvider.class,
+        ConfigFileAuthenticationDetailsProvider.class
+})
+@BootstrapContextCompatible
+public class OracleCloudVaultConfigurationClient implements ConfigurationClient {
+
+    private static final Logger LOG = LoggerFactory.getLogger(OracleCloudVaultConfigurationClient.class);
+
+    private final OracleCloudVaultClientConfiguration oracleCloudVaultClientConfiguration;
+    private final ApplicationConfiguration applicationConfiguration;
+    private final ExecutorService executorService;
+    private final SecretsClient secretsClient;
+    private final VaultsClient vaultsClient;
+
+    /**
+     * Default Constructor.
+     *
+     * @param oracleCloudVaultClientConfiguration   Oracle CloudVault Client Configuration
+     * @param applicationConfiguration              The application configuration
+     * @param executorService                       Executor Service
+     * @throws Exception                            If no configuration is provided
+     */
+    public OracleCloudVaultConfigurationClient(OracleCloudVaultClientConfiguration oracleCloudVaultClientConfiguration,
+                                    ApplicationConfiguration applicationConfiguration,
+                                    @Named(TaskExecutors.IO) @Nullable ExecutorService executorService) throws Exception {
+        this.oracleCloudVaultClientConfiguration = oracleCloudVaultClientConfiguration;
+        this.applicationConfiguration = applicationConfiguration;
+        this.executorService = executorService;
+
+        BasicAuthenticationDetailsProvider provider = null;
+        if (oracleCloudVaultClientConfiguration.isUseInstancePrincipal()) {
+            provider = InstancePrincipalsAuthenticationDetailsProvider.builder().build();
+        } else {
+            try {
+                provider = new ConfigFileAuthenticationDetailsProvider(oracleCloudVaultClientConfiguration.getPathToConfig(), oracleCloudVaultClientConfiguration.getProfile());
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
Could you log this exception instead of dumping the stack trace?

Reply:
Fixed in https://github.com/micronaut-projects/micronaut-core/pull/3045/commits/a2b03f3d41e66db97edc2a67ee028a7fb1ae8d38